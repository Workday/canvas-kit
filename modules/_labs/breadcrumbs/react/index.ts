import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
