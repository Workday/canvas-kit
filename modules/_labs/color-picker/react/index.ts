export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
