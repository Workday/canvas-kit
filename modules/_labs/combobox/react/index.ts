import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
