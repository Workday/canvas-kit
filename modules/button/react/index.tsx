import Button from './lib/Button';

export default Button;
export {default as Button, beta_Button, ButtonProps} from './lib/Button';
export {default as IconButton, IconButtonProps} from './lib/IconButton';
export {default as DropdownButton} from './lib/DropdownButton';
export {default as TextButton, TextButtonProps} from './lib/TextButton';
export {
  default as IconButtonToggleGroup,
  IconButtonToggleGroupProps,
} from './lib/IconButtonToggleGroup';
export * from './lib/types';
