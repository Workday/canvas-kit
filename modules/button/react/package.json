{
  "name": "@workday/canvas-kit-react-button",
  "version": "4.0.0-beta.5",
  "description": "A collection of various Canvas Buttons",
  "author": "Workday, Inc. (https://www.workday.com)",
  "license": "Apache-2.0",
  "main": "dist/commonjs/index.js",
  "module": "dist/es6/index.js",
  "sideEffects": false,
  "types": "dist/es6/index.d.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/Workday/canvas-kit/tree/master/modules/button/react"
  },
  "files": [
    "dist/",
    "lib/",
    "index.ts",
    "ts3.5/**/*"
  ],
  "typesVersions": {
    "<=3.5": {
      "*": [
        "ts3.5/*"
      ]
    }
  },
  "scripts": {
    "watch": "yarn build:es6 -w",
    "clean": "rimraf dist && rimraf ts3.5 && rimraf .build-info && mkdirp dist && mkdirp ts3.5/dist",
    "build:cjs": "tsc -p tsconfig.cjs.json",
    "build:es6": "tsc -p tsconfig.es6.json",
    "build:rebuild": "npm-run-all clean build",
    "build:downlevel-dts": "yarn run downlevel-dts dist ts3.5/dist",
    "build": "npm-run-all --parallel build:cjs build:es6 --sequential build:downlevel-dts",
    "depcheck": "node ../../../utils/check-dependencies-exist.js"
  },
  "keywords": [
    "canvas",
    "canvas-kit",
    "react",
    "components",
    "workday",
    "button"
  ],
  "peerDependencies": {
    "react": ">= 16.8 < 17"
  },
  "dependencies": {
    "@emotion/core": "^10.0.28",
    "@emotion/is-prop-valid": "^0.8.2",
    "@emotion/styled": "^10.0.27",
    "@workday/canvas-kit-labs-react-core": "4.0.0-beta.5",
    "@workday/canvas-kit-react-common": "4.0.0-beta.5",
    "@workday/canvas-kit-react-core": "4.0.0-beta.5",
    "@workday/canvas-kit-react-icon": "4.0.0-beta.5",
    "@workday/canvas-system-icons-web": "^1.0.20",
    "@workday/design-assets-types": "^0.2.4"
  }
}
