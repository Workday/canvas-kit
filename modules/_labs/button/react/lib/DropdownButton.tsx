import * as React from 'react';
import {caretDownIcon} from '@workday/canvas-system-icons-web';
import {GrowthBehavior} from '@workday/canvas-kit-react-common';
import {DropdownButtonVariant, ButtonSize, ButtonIconPosition} from './types';
import {ButtonContainer, ButtonLabel, ButtonLabelIcon} from './parts';
import {buttonColors} from './Button';

export interface DropdownButtonProps
  extends React.HTMLAttributes<HTMLButtonElement>,
    GrowthBehavior {
  /**
   * The variant of the Button.
   * @default DropdownButtonVariant.Secondary
   */
  variant: DropdownButtonVariant;
  /**
   * The size of the Button.
   * @default ButtonSize.Medium
   */
  size: ButtonSize.Medium | ButtonSize.Large;
  /**
   * The ref to the button that the styled component renders.
   */
  buttonRef?: React.Ref<HTMLButtonElement>;
}

const DropdownButton = (props: DropdownButtonProps) => {
  const {
    variant = DropdownButtonVariant.Secondary,
    size = ButtonSize.Medium,
    buttonRef,
    children,
    ...elemProps
  } = props;
  return (
    <ButtonContainer
      colors={buttonColors[variant]}
      size={size}
      buttonRef={buttonRef}
      {...elemProps}
    >
      <ButtonLabel>{children}</ButtonLabel>
      <ButtonLabelIcon
        size={size}
        icon={caretDownIcon}
        iconPosition={ButtonIconPosition.Right}
        dropdown={true}
      />
    </ButtonContainer>
  );
};

DropdownButton.Variant = DropdownButtonVariant;
DropdownButton.Size = {
  Medium: ButtonSize.Medium,
  Large: ButtonSize.Large,
};

export default DropdownButton;
