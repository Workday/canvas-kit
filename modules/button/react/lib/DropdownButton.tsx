import * as React from 'react';
import {caretDownIcon} from '@workday/canvas-system-icons-web';
import {GrowthBehavior, useTheme, Themeable} from '@workday/canvas-kit-react-common';
import {DropdownButtonVariant, ButtonIconPosition, ButtonOrAnchorComponent} from './types';
import {ButtonContainer, ButtonLabel, ButtonLabelIcon} from './parts';
import {getButtonColors} from './Button';

export interface DropdownButtonProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    Themeable,
    GrowthBehavior {
  /**
   * The variant of the Button.
   * @default DropdownButtonVariant.Secondary
   */
  variant?: DropdownButtonVariant;
  /**
   * The size of the Button.
   * @default 'medium'
   */
  size?: 'medium' | 'large';
  /**
   * The ref to the button that the styled component renders.
   */
  buttonRef?: React.Ref<HTMLButtonElement>;
  /**
   * The alternative container type for the button. Uses Emotion's special `as` prop.
   * Will render an `a` tag instead of a `button` when defined.
   */
  as?: 'a';
}

const DropdownButton: ButtonOrAnchorComponent<
  DropdownButtonProps,
  typeof DropdownButtonVariant
> = ({
  // TODO: Fix useTheme and make it a real hook
  // eslint-disable-next-line react-hooks/rules-of-hooks
  theme = useTheme(),
  variant = DropdownButtonVariant.Secondary,
  size = 'medium',
  buttonRef,
  children,
  ...elemProps
}: DropdownButtonProps) => (
  <ButtonContainer
    colors={getButtonColors(variant, theme)}
    size={size}
    ref={buttonRef}
    {...elemProps}
  >
    <ButtonLabel>{children}</ButtonLabel>
    <ButtonLabelIcon
      size={size}
      icon={caretDownIcon}
      iconPosition={ButtonIconPosition.Right}
      dropdown={true}
    />
  </ButtonContainer>
);

DropdownButton.Variant = DropdownButtonVariant;
DropdownButton.Size = {
  Medium: 'medium',
  Large: 'large',
} as const;

export default DropdownButton;
