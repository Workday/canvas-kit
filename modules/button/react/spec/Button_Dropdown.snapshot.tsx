import * as React from 'react';
import * as renderer from 'react-test-renderer';
import DropdownButton from '../lib/DropdownButton';

describe('Button (Dropdown) Snapshots', () => {
  test('renders a large, dropdown primary button', () => {
    const component = renderer.create(
      <DropdownButton size={DropdownButton.Size.Large} variant={DropdownButton.Variant.Primary}>
        Button
      </DropdownButton>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders a medium, dropdown primary button', () => {
    const component = renderer.create(
      <DropdownButton size={DropdownButton.Size.Medium} variant={DropdownButton.Variant.Primary}>
        Button
      </DropdownButton>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders a large, dropdown secondary button', () => {
    const component = renderer.create(
      <DropdownButton size={DropdownButton.Size.Large} variant={DropdownButton.Variant.Secondary}>
        Button
      </DropdownButton>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders a medium, dropdown secondary button', () => {
    const component = renderer.create(
      <DropdownButton size={DropdownButton.Size.Medium} variant={DropdownButton.Variant.Secondary}>
        Button
      </DropdownButton>
    );
    expect(component).toMatchSnapshot();
  });
});
