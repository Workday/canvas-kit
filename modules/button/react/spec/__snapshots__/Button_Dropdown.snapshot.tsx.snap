// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button (Dropdown) Snapshots renders a large, dropdown primary button 1`] = `
.emotion-5 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
  min-width: 112px;
  height: 48px;
  padding: 0 20px;
}

.emotion-5:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-5:disabled,
.emotion-5:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-5 span .wd-icon-fill,
.emotion-5 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #ffffff;
}

.emotion-5:focus {
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
}

.emotion-5:focus span .wd-icon-fill,
.emotion-5:focus span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-5:hover:focus {
  background-color: #005cb9;
}

.emotion-5:active,
.emotion-5:focus:active,
.emotion-5:hover:active {
  background-color: #004387;
  border-color: transparent;
  color: #ffffff;
}

.emotion-5:active span .wd-icon-fill,
.emotion-5:focus:active span .wd-icon-fill,
.emotion-5:hover:active span .wd-icon-fill,
.emotion-5:active span .wd-icon-accent,
.emotion-5:focus:active span .wd-icon-accent,
.emotion-5:hover:active span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-5:hover {
  background-color: #005cb9;
  border-color: transparent;
  color: #ffffff;
}

.emotion-5:hover span .wd-icon-fill,
.emotion-5:hover span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-5:disabled,
.emotion-5:active:disabled,
.emotion-5:focus:disabled,
.emotion-5:hover:disabled {
  background-color: #a6d2ff;
  border-color: transparent;
  color: #ffffff;
}

.emotion-5:disabled span .wd-icon-fill,
.emotion-5:active:disabled span .wd-icon-fill,
.emotion-5:focus:disabled span .wd-icon-fill,
.emotion-5:hover:disabled span .wd-icon-fill,
.emotion-5:disabled span .wd-icon-accent,
.emotion-5:active:disabled span .wd-icon-accent,
.emotion-5:focus:disabled span .wd-icon-accent,
.emotion-5:hover:disabled span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-5:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-5:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 16px;
  padding: 0 12px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 8px 0 0;
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-2 svg {
  width: 24px;
  height: 24px;
}

<button
  className="emotion-5 emotion-6"
>
  <span
    className="emotion-0 emotion-1"
  >
    Button
  </span>
  <span
    className="emotion-3 emotion-4"
  >
    <span
      className="emotion-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-caret-down wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M11.727 16.59c.15.226.398.222.546 0l5.454-8.18c.15-.226.044-.41-.227-.41h-11c-.276 0-.375.188-.227.41l5.454 8.18z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </span>
</button>
`;

exports[`Button (Dropdown) Snapshots renders a large, dropdown secondary button 1`] = `
.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 16px;
  padding: 0 12px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 8px 0 0;
}

.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-2 svg {
  width: 24px;
  height: 24px;
}

.emotion-5 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
  min-width: 112px;
  height: 48px;
  padding: 0 20px;
}

.emotion-5:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-5:disabled,
.emotion-5:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-5 span .wd-icon-fill,
.emotion-5 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-5 .button-label-data {
  color: #333333;
}

.emotion-5:focus {
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
}

.emotion-5:focus span .wd-icon-fill,
.emotion-5:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-5:hover:focus {
  background-color: #dfe2e6;
}

.emotion-5:active,
.emotion-5:focus:active,
.emotion-5:hover:active {
  background-color: #ced3d9;
  border-color: transparent;
  color: #333333;
}

.emotion-5:active span .wd-icon-fill,
.emotion-5:focus:active span .wd-icon-fill,
.emotion-5:hover:active span .wd-icon-fill,
.emotion-5:active span .wd-icon-accent,
.emotion-5:focus:active span .wd-icon-accent,
.emotion-5:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-5:hover {
  background-color: #dfe2e6;
  border-color: transparent;
  color: #333333;
}

.emotion-5:hover span .wd-icon-fill,
.emotion-5:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-5:disabled,
.emotion-5:active:disabled,
.emotion-5:focus:disabled,
.emotion-5:hover:disabled {
  background-color: #f6f7f8;
  border-color: transparent;
  color: #a1aab3;
}

.emotion-5:disabled span .wd-icon-fill,
.emotion-5:active:disabled span .wd-icon-fill,
.emotion-5:focus:disabled span .wd-icon-fill,
.emotion-5:hover:disabled span .wd-icon-fill,
.emotion-5:disabled span .wd-icon-accent,
.emotion-5:active:disabled span .wd-icon-accent,
.emotion-5:focus:disabled span .wd-icon-accent,
.emotion-5:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-5:disabled .button-label-data,
.emotion-5:active:disabled .button-label-data,
.emotion-5:focus:disabled .button-label-data,
.emotion-5:hover:disabled .button-label-data {
  color: #a1aab3;
}

.emotion-5:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-5:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

<button
  className="emotion-5 emotion-6"
>
  <span
    className="emotion-0 emotion-1"
  >
    Button
  </span>
  <span
    className="emotion-3 emotion-4"
  >
    <span
      className="emotion-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-caret-down wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M11.727 16.59c.15.226.398.222.546 0l5.454-8.18c.15-.226.044-.41-.227-.41h-11c-.276 0-.375.188-.227.41l5.454 8.18z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </span>
</button>
`;

exports[`Button (Dropdown) Snapshots renders a medium, dropdown primary button 1`] = `
.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-2 svg {
  width: 24px;
  height: 24px;
}

.emotion-5 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
  min-width: 96px;
  height: 40px;
  padding: 0 16px;
}

.emotion-5:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-5:disabled,
.emotion-5:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-5 span .wd-icon-fill,
.emotion-5 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #ffffff;
}

.emotion-5:focus {
  background-color: #0875e1;
  border-color: transparent;
  color: #ffffff;
}

.emotion-5:focus span .wd-icon-fill,
.emotion-5:focus span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-5:hover:focus {
  background-color: #005cb9;
}

.emotion-5:active,
.emotion-5:focus:active,
.emotion-5:hover:active {
  background-color: #004387;
  border-color: transparent;
  color: #ffffff;
}

.emotion-5:active span .wd-icon-fill,
.emotion-5:focus:active span .wd-icon-fill,
.emotion-5:hover:active span .wd-icon-fill,
.emotion-5:active span .wd-icon-accent,
.emotion-5:focus:active span .wd-icon-accent,
.emotion-5:hover:active span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-5:hover {
  background-color: #005cb9;
  border-color: transparent;
  color: #ffffff;
}

.emotion-5:hover span .wd-icon-fill,
.emotion-5:hover span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-5:disabled,
.emotion-5:active:disabled,
.emotion-5:focus:disabled,
.emotion-5:hover:disabled {
  background-color: #a6d2ff;
  border-color: transparent;
  color: #ffffff;
}

.emotion-5:disabled span .wd-icon-fill,
.emotion-5:active:disabled span .wd-icon-fill,
.emotion-5:focus:disabled span .wd-icon-fill,
.emotion-5:hover:disabled span .wd-icon-fill,
.emotion-5:disabled span .wd-icon-accent,
.emotion-5:active:disabled span .wd-icon-accent,
.emotion-5:focus:disabled span .wd-icon-accent,
.emotion-5:hover:disabled span .wd-icon-accent {
  fill: #ffffff;
}

.emotion-5:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-5:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0 8px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 4px 0 0;
}

<button
  className="emotion-5 emotion-6"
>
  <span
    className="emotion-0 emotion-1"
  >
    Button
  </span>
  <span
    className="emotion-3 emotion-4"
  >
    <span
      className="emotion-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-caret-down wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M11.727 16.59c.15.226.398.222.546 0l5.454-8.18c.15-.226.044-.41-.227-.41h-11c-.276 0-.375.188-.227.41l5.454 8.18z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </span>
</button>
`;

exports[`Button (Dropdown) Snapshots renders a medium, dropdown secondary button 1`] = `
.emotion-2 {
  display: inline-block;
}

.emotion-2 svg {
  display: block;
}

.emotion-2 .wd-icon-fill {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-fill {
  fill: #333d47;
}

.emotion-2 .wd-icon-accent {
  fill: #7b858f;
}

.emotion-2:hover .wd-icon-accent {
  fill: #333d47;
}

.emotion-2 .wd-icon-background {
  fill: transparent;
}

.emotion-2:hover .wd-icon-background {
  fill: transparent;
}

.emotion-2 svg {
  width: 24px;
  height: 24px;
}

.emotion-0 {
  position: relative;
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  font-weight: 700;
  font-family: "Roboto","Helvetica Neue","Helvetica","Arial",sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-size: 14px;
  padding: 0 8px;
}

.emotion-0:hover:active {
  background-color: transparent;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 4px 0 0;
}

.emotion-5 {
  box-sizing: border-box;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 14px;
  border-radius: 999px;
  border: 2px solid transparent;
  box-shadow: none;
  position: relative;
  cursor: pointer;
  outline: none;
  -webkit-transition: all 120ms linear;
  transition: all 120ms linear;
  vertical-align: middle;
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
  min-width: 96px;
  height: 40px;
  padding: 0 16px;
}

.emotion-5:hover:active {
  -webkit-transition-duration: 40ms;
  transition-duration: 40ms;
}

.emotion-5:disabled,
.emotion-5:disabled:active {
  cursor: default;
  box-shadow: none;
}

.emotion-5 span .wd-icon-fill,
.emotion-5 span .wd-icon-accent {
  -webkit-transition: fill 120ms ease-in;
  transition: fill 120ms ease-in;
  fill: #7b858f;
}

.emotion-5 .button-label-data {
  color: #333333;
}

.emotion-5:focus {
  background-color: #f0f1f2;
  border-color: transparent;
  color: #333333;
}

.emotion-5:focus span .wd-icon-fill,
.emotion-5:focus span .wd-icon-accent {
  fill: #7b858f;
}

.emotion-5:hover:focus {
  background-color: #dfe2e6;
}

.emotion-5:active,
.emotion-5:focus:active,
.emotion-5:hover:active {
  background-color: #ced3d9;
  border-color: transparent;
  color: #333333;
}

.emotion-5:active span .wd-icon-fill,
.emotion-5:focus:active span .wd-icon-fill,
.emotion-5:hover:active span .wd-icon-fill,
.emotion-5:active span .wd-icon-accent,
.emotion-5:focus:active span .wd-icon-accent,
.emotion-5:hover:active span .wd-icon-accent {
  fill: #333d47;
}

.emotion-5:hover {
  background-color: #dfe2e6;
  border-color: transparent;
  color: #333333;
}

.emotion-5:hover span .wd-icon-fill,
.emotion-5:hover span .wd-icon-accent {
  fill: #333d47;
}

.emotion-5:disabled,
.emotion-5:active:disabled,
.emotion-5:focus:disabled,
.emotion-5:hover:disabled {
  background-color: #f6f7f8;
  border-color: transparent;
  color: #a1aab3;
}

.emotion-5:disabled span .wd-icon-fill,
.emotion-5:active:disabled span .wd-icon-fill,
.emotion-5:focus:disabled span .wd-icon-fill,
.emotion-5:hover:disabled span .wd-icon-fill,
.emotion-5:disabled span .wd-icon-accent,
.emotion-5:active:disabled span .wd-icon-accent,
.emotion-5:focus:disabled span .wd-icon-accent,
.emotion-5:hover:disabled span .wd-icon-accent {
  fill: #b9c0c7;
}

.emotion-5:disabled .button-label-data,
.emotion-5:active:disabled .button-label-data,
.emotion-5:focus:disabled .button-label-data,
.emotion-5:hover:disabled .button-label-data {
  color: #a1aab3;
}

.emotion-5:not([disabled]):focus {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

.emotion-5:not([disabled]):active {
  border-color: transparent;
  -webkit-animation: animation-1h1pob6 100ms;
  animation: animation-1h1pob6 100ms;
  box-shadow: 0 0 0 2px #ffffff,0 0 0 4px #0875e1;
}

<button
  className="emotion-5 emotion-6"
>
  <span
    className="emotion-0 emotion-1"
  >
    Button
  </span>
  <span
    className="emotion-3 emotion-4"
  >
    <span
      className="emotion-2"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"24\\" height=\\"24\\" class=\\"wd-icon-caret-down wd-icon\\" focusable=\\"false\\" role=\\"presentation\\" viewBox=\\"0 0 24 24\\"><g fill-rule=\\"evenodd\\" class=\\"wd-icon-container\\"><path d=\\"M11.727 16.59c.15.226.398.222.546 0l5.454-8.18c.15-.226.044-.41-.227-.41h-11c-.276 0-.375.188-.227.41l5.454 8.18z\\" class=\\"wd-icon-fill\\"/></g></svg>",
        }
      }
    />
  </span>
</button>
`;
